omisego-sdk/src/main/java/co/omisego/omisego/network/ewallet/EWalletClient.kt

```diff
@@ -0,0 +1,109 @@
+package co.omisego.omisego.network.ewallet
+
+import co.omisego.omisego.constant.Exceptions
+import co.omisego.omisego.custom.JsonConverterFactory
+import co.omisego.omisego.network.InterceptorProvider
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+
+/**
+ * OmiseGO
+ *
+ * Created by Phuchit Sirimongkolsathien on 11/10/2017 AD.
+ * Copyright © 2017-2018 OmiseGO. All rights reserved.
+ */
+
+/**
+ * The class EWalletClient represents an object that knows how to interact with OmiseGO API.
+ *
+ * Create instances using [EWalletClient.Builder] and pass your implementation of [Callback<T>] interface
+ * to generate an implementation
+ *
+ * For example,
+ * <code>
+ * val eWalletClient = EWalletClient.Builder {
+ *      authenticationToken = YOUR_TOKEN
+ *      baseURL = YOUR_BASE_URL
+ * }.build()
+ *
+ * </code>
+ *
+ */
+class EWalletClient {
+    internal lateinit var eWalletAPI: EWalletAPI
+
+    /**
+     * Build a new [EWalletClient].
+     * Set [authenticationToken] and [baseURL] are required before calling [Builder.build].
+     * Set [debug] true for printing a log
+     *
+     * @receiver A [Builder]'s methods.
+     */
+    class Builder(init: Builder.() -> Unit) {
+        var debug: Boolean = false
+
+        /**
+         * Set the API [authenticationToken].
+         * The [authenticationToken] should be "OMG Base64(api_key:authentication_token)"
+         */
+        var authenticationToken: String = ""
+            set(value) {
+                if (value.isEmpty()) throw Exceptions.emptyAuthenticationToken
+                field = value
+            }
+
+        /**
+         * Set the URL of the OmiseGO Wallet API [baseURL].
+         */
+        var baseURL: String = ""
+            set(value) {
+                if (value.isEmpty()) throw Exceptions.emptyBaseURL
+                field = value
+            }
+
+
+        /**
+         * For testing purpose
+         */
+        internal var debugURL: HttpUrl? = null
```

Comment:
Any reason not to use a `String` like `baseURL` ?
Same thing here for the naming, OkHttp's methods are `baseUrl`, here fields are `baseURL`.
Just a detail to note :)

Reply:
I'm using it to stub the `MockWebServer` HttpUrl. The string isn't worked :(

See the reference: https://github.com/square/okhttp/tree/master/mockwebserver